std/async/pool.ts

```diff
@@ -27,20 +32,39 @@ export function pooledMap<T, R>(
   (async (): Promise<void> => {
     const writer = res.writable.getWriter();
     const executing: Array<Promise<unknown>> = [];
-    for await (const item of array) {
-      const p = Promise.resolve().then(() => iteratorFn(item));
-      writer.write(p);
-      const e: Promise<unknown> = p.then(() =>
-        executing.splice(executing.indexOf(e), 1)
-      );
-      executing.push(e);
-      if (executing.length >= poolLimit) {
-        await Promise.race(executing);
+    try {
+      for await (const item of array) {
+        const p = Promise.resolve().then(() => iteratorFn(item));
+        // Only write on success. If we `writer.write()` a rejected promise,
+        // that will end the iteration. We don't want that yet. Instead let it
+        // fail the race, taking us to the catch block where all currently
+        // executing jobs are allowed to finish and all rejections among them
+        // can be reported together.
+        p.then((v) => writer.write(Promise.resolve(v))).catch(() => {});
+        const e: Promise<unknown> = p.then(() =>
+          executing.splice(executing.indexOf(e), 1)
+        );
+        executing.push(e);
+        if (executing.length >= poolLimit) {
+          await Promise.race(executing);
+        }
+      }
+      // Wait until all ongoing events have processed, then close the writer.
+      await Promise.all(executing);
+      writer.close();
```

Comment:
how about doing this in `finally`?

Reply:
It closes on error, so `writer.close()` will throw if done in the catch branch.